src/Compilers/Core/Portable/Syntax/SyntaxToken.cs

```diff
@@ -591,6 +591,26 @@ public SyntaxToken GetNextToken(bool includeZeroWidth = false, bool includeSkipp
             return SyntaxNavigator.Instance.GetNextToken(this, includeZeroWidth, includeSkipped, includeDirectives, includeDocumentationComments);
         }
 
+        /// <summary>
+        /// Sibling nodes and tokens which follow this token, where each element is a SyntaxNodeOrToken instance.
+        /// </summary>
+        internal ChildSyntaxList.Segment SiblingsAfterSelf()
+        {
+            Debug.Assert(Parent is object);
+            Debug.Assert(this == ChildSyntaxList.ItemInternal(Parent, Index));
+            return new ChildSyntaxList.Segment(Parent!, Index);
+        }
+
+        /// <summary>
+        /// Siblings which precede this token, where each element is a SyntaxNodeOrToken instance.
+        /// </summary>
+        internal ChildSyntaxList.Reversed SiblingsBeforeSelf()
+        {
+            Debug.Assert(Parent is object);
+            Debug.Assert(this == ChildSyntaxList.ItemInternal(Parent, Index));
+            return new ChildSyntaxList.Reversed(Parent!, Index);
+        }
```

Comment:
this is more a question, versus a statement:

It looks like SiblingsAfterSelf/SiblingsBeforeSelf are used to just get the very next non-None sibling after/before this entity.  they're never needed for anything beyond that.  So why not just make this:

`internal SyntaxNodeOrToken SiblingAfterSelf()` (and same for teh 'before' version)?

do you really need more than that ability?

Reply:
allowing further enumeration is required to support predicates and tokens w/ a raw kind of `None`,
like when we don't return in [this region of code](/dotnet/roslyn/pull/69740/files/bef17ddf5a0fdb96513596e545b5969e22389420#diff-8a0addb381c7608eb2dbc3662640f660f5afe9b8568606b1d11c6296405f1ab8R450-R454)